app/controllers/orders_controller.rb

```diff
@@ -99,4 +105,32 @@ def check_current_user
     flash[:danger] = t "something_wrong"
     redirect_to user_orders_path current_user
   end
+
+  def find_order
+    @order = Order.find_by id: params[:id]
+
+    return if @order
+
+    flash[:danger] = t "something_wrong"
+    redirect_to user_orders_path current_user
+  end
+
+  def can_destroy?
+    if @order.status_id == Settings.status.pendding
+      true
+    elsif @order.status_id == Settings.status.approved
+      get_status_destroy_for_approved
+    else
+      false
+    end
+  end
+
+  def get_status_destroy_for_approved
+    if ((DateTime.now.to_time - @order.created_at.to_time) / 1.hour).to_i < 24
+      true
+    else
+      @message = t "message_cannot_destroy_order"
+      false
+    end
```

Comment:
```suggestion
    DateTime.now - @order.created_at.to_time < 24.hours
```
xử lý true/false ở `can_destroy?` để create message em ơi

Reply:
Em có 4 trạng thái anh ơi , cái can_destroy? của em để check 1 là pendding, 2 là các trạng thái khác trừ approved. Còn cái hàm get_status_for_approved để e check xem đã quá 24h chưa anh ạ, E để luôn trong kia thì k qua được rubocop, e tách ra thì chưa biết cách nào để ngắn hơn, anh có suggest gì k a =)))

Reply:
rubocop báo gì ?

Reply:
rubocop báo high condition anh à